lib/SPIRV/SPIRVLowerMemmove.cpp

```diff
@@ -111,13 +106,38 @@ class SPIRVLowerMemmoveBase : public InstVisitor<SPIRVLowerMemmoveBase> {
     I.dropAllReferences();
     I.eraseFromParent();
   }
+  bool expandMemIntrinsicUses(Function &F) {
+    bool Changed = false;
+
+    for (auto I = F.user_begin(), E = F.user_end(); I != E;) {
+      MemMoveInst *Inst = cast<MemMoveInst>(*I);
+      ++I;
```

Comment:
Perhaps worth putting a comment why the increment is done here, and not as part of the for-statement.

Reply:
I'm not sure it is needed, I replaced this loop with `for(User *U : ... `

Reply:
That's also a solution.  But perhaps you need to use `make_early_inc_range(F.users())` then, as the loop may modify the range it is iterating over?

Reply:
I think you are right, thanks. It seems `eraseFromParent` deletes instruction and can invalidate the range.